src/main/kotlin/org/jitsi/nlj/rtp/codec/vp9/Vp9Packet.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.nlj.rtp.codec.vp9
+
+import org.jitsi.nlj.rtp.ParsedVideoPacket
+import org.jitsi.utils.logging2.cwarn
+import org.jitsi.rtp.extensions.bytearray.hashCodeOfSegment
+import org.jitsi.utils.logging2.createLogger
+import org.jitsi_modified.impl.neomedia.codec.video.vp9.DePacketizer
+import kotlin.properties.Delegates
+
+/**
+ * If this [Vp9Packet] instance is being created via a clone,
+ * we've already parsed the packet itself and determined whether
+ * or not its a keyframe and what its spatial layer index is,
+ * so the constructor allows passing in those values if
+ * they're already known.  If they're null, this instance
+ * will do the parsing itself.
+ */
+class Vp9Packet private constructor (
+    buffer: ByteArray,
+    offset: Int,
+    length: Int,
+    isKeyframe: Boolean?,
+    isStartOfFrame: Boolean?,
+    encodingIndex: Int?,
+    height: Int?,
+    pictureId: Int?,
+    TL0PICIDX: Int?
+) : ParsedVideoPacket(buffer, offset, length, encodingIndex) {
+
+    constructor(
+        buffer: ByteArray,
+        offset: Int,
+        length: Int
+    ) : this(buffer, offset, length,
+        isKeyframe = null,
+        isStartOfFrame = null,
+        encodingIndex = null,
+        height = null,
+        pictureId = null,
+        TL0PICIDX = null
+    )
+
+    /** Due to the format of the VP9 payload, this value is only reliable for packets where [isStartOfFrame] is true. */
```

Comment:
I wonder if we should enforce this somehow, like by returning an `Optional<Boolean>` or something?

Reply:
Actually this comment was copied from the VP8 packet, and isn't true for VP9...it's maybe worth looking at for VP8 though.